app/assets/javascripts/dashboard/base.js

```diff
@@ -0,0 +1,28 @@
+DashboardBase = function() {
+  var onCLickJSAddMetadataField = function() {
+    $('form').on('click', '.js-add-metadata-fields', function(e) {
+      e.preventDefault();
+      blueprint = $(e.target).data("blueprint");
+
+      var association = $(e.target).data("association");
+      var new_id = new Date().getTime();
+      var regexp = new RegExp("new_" + association, "g");
+      $('.js-metadata-container').append(blueprint.replace(regexp, new_id));
+    });
+  }
+
+  var onClickJSRemoveMetadataField = function() {
+    $('form').on('click', '.js-remove-metadata-fields',
+    function(e) {
+      parent = $(this).closest('.js-metadata-fields')
+      parent.find('input').val('');
+      parent.removeClass('d-flex');
+      parent.hide();
+    })
+  }
+
+  this.init = function() {
+    onCLickJSAddMetadataField();
+    onClickJSRemoveMetadataField();
```

Comment:
Should we better to have a class to handle metadata field base on data behavior attribute.

new MetaDataFields($("[data-behavior='metadata-fields']"))

Reply:
i'm not really sure what you want me to do here.